konnectivity-client/pkg/client/client.go

```diff
@@ -35,7 +35,7 @@ import (
 type Tunnel interface {
 	// Dial connects to the address on the named network, similar to
 	// what net.Dial does. The only supported protocol is tcp.
-	Dial(protocol, address string) (net.Conn, error)
+	DialContext(ctx context.Context, protocol, address string) (net.Conn, error)
```

Comment:
This is a breaking interface change. Would it make sense to keep both interfaces here for backwards compatibility? The golang net dialer has both functions: `net.Dial` and `net.DialContext` 

cc: @cheftako

Reply:
This is a breaking change. Technically as a v0 release that would be allowed but certainly not preferred. As this is the piece which is wired into the KAS I'm less concerned about it. If we take this it means that whoever integrates this fix into K/K has to fix all the callers of this client. On the plus side it means that whoever integrates this fix into K/K has to fix ALL the callers of this client. 

I'm much more concerned with ensuring we remain backward compatible with the protocol/transport between the KAS and the Network Proxy-Server. The Network Proxy Server is a reference implementation and we expect that other groups can swap it out. So if we make a backward incompatible change there it isn't something we can fix.

Reply:
the current interface is a hang landmine... if we're still in v0 we should change the signature to require a context to ensure this doesn't hang when the parent request has moved on

Reply:
I also prefer making the client strict since the protocol will remain compatible, and there are extremely few call sites to adapt.